packages/nextjs-mf/src/plugins/NextFederationPlugin/index.ts

```diff
@@ -0,0 +1,176 @@
+import { removeUnnecessarySharedKeys } from './remove-unnecessary-shared-keys';
+import { setOptions } from './set-options';
+import {
+  validateCompilerOptions,
+  validatePluginOptions,
+} from './validate-options';
+import {
+  applyServerPlugins,
+  configureServerCompilerOptions,
+  configureServerLibraryAndFilename,
+  handleServerExternals,
+} from './apply-server-plugins';
+import { applyClientPlugins } from './apply-client-plugins';
+
+/**
+ * NextFederationPlugin is a webpack plugin that handles Next.js application
+ * federation using Module Federation.
+ */
+export class NextFederationPlugin {
+  _options: ModuleFederationPluginOptions;
+  _extraOptions: NextFederationPluginExtraOptions;
+
+  /**
+   * Constructs the NextFederationPlugin with the provided options.
+   *
+   * @param options The options to configure the plugin.
+   */
+  constructor(options: NextFederationPluginOptions) {
+    const { mainOptions, extraOptions } = setOptions(options);
+    this._options = mainOptions;
+    this._extraOptions = extraOptions;
+  }
+
+  apply(compiler: Compiler) {
+    // Validate the compiler options
+    const validCompile = validateCompilerOptions(compiler);
+    if (!validCompile) return;
+    // Validate the NextFederationPlugin options
+    validatePluginOptions(this._options);
+
+    // Check if the compiler is for the server or client
+    const isServer = compiler.options.name === 'server';
+    const { webpack } = compiler;
+
+    // Apply the CopyFederationPlugin
+    new CopyFederationPlugin(isServer).apply(compiler);
+
+    // If remotes are provided, parse them
+    if (this._options.remotes) {
+      // @ts-ignore
+      this._options.remotes = parseRemotes(this._options.remotes);
+    }
+
+    // If shared modules are provided, remove unnecessary shared keys from the default share scope
+    if (this._options.shared) {
+      removeUnnecessarySharedKeys(this._options.shared as SharedObject);
+    }
+
+    const ModuleFederationPlugin: container.ModuleFederationPlugin =
+      getModuleFederationPluginConstructor(isServer, compiler);
+
+    const defaultShared = retrieveDefaultShared(isServer);
+    if (isServer) {
+      // Refactored server condition
+      configureServerCompilerOptions(compiler);
+      configureServerLibraryAndFilename(this._options);
+
+      applyServerPlugins(compiler, this._options);
+      handleServerExternals(compiler, {
+        ...this._options,
+        shared: { ...defaultShared, ...this._options.shared },
+      });
+    } else {
+      applyClientPlugins(compiler, this._options, this._extraOptions);
+    }
+
+    applyPathFixes(compiler, this._extraOptions);
+
+    // @ts-ignore
+    const hostFederationPluginOptions: ModuleFederationPluginOptions = {
+      ...this._options,
+      runtime: false,
+      exposes: {
+        //something must be exposed in order to generate a remote entry, which is needed to kickstart runtime
+        './noop': require.resolve('../../federation-noop.ts'),
```

Comment:
```suggestion
        './noop': require.resolve('../../federation-noop'),
```
I think this line shouldn't try to resolve a `.ts` extension like the other `require.resolve` calls in 4c7b3ffad0726c6d9f51a8fb72ec7eefb3633073.

I've just tried to upgrade to the latest rc (6.7.2-rc.1) and the dev server is immediately crashing with the following error:
```
Error: Cannot find module '../../federation-noop.ts'
Require stack:
- /SOME_PATH/node_modules/@module-federation/nextjs-mf/src/plugins/NextFederationPlugin/index.js
- /SOME_PATH/node_modules/@module-federation/nextjs-mf/src/index.js
```

Reply:
thanks <3